src/mixins/openLocal.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * @copyright Copyright (c) 2023 Julius Härtl <jus@bitgrid.net>
+ *
+ * @author Julius Härtl <jus@bitgrid.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { getRootUrl } from '@nextcloud/router'
+import axios from '@nextcloud/axios'
+import { getNextcloudUrl } from '../helpers/url.js'
+import { getCurrentUser } from '@nextcloud/auth'
+
+// FIXME: Migrate to vue component
+export default {
+	data() {
+		return {
+			openingLocally: false,
+		}
+	},
+	methods: {
+		unlockAndOpenLocally() {
+			if (this.openingLocally) {
+				this.unlockFile()
+					.catch(_ => {}) // Unlocking failed, possibly because file was not locked, we want to proceed regardless.
```

Comment:
Maybe check the server response more carefully?
This might be anything from dns failure to 500, 502. Not sure what the response would be in case the file was not locked. But I think the catch should be restricted to the case you are anticipating.